torch/_refs/__init__.py

```diff
@@ -5431,12 +5431,12 @@ def masked_fill(a: TensorLikeType, mask: TensorLikeType, value: TensorOrNumberLi
             value_ndim == 0,
             lambda: f"only supports a 0-dimensional value tensor, but got tensor with {value_ndim} dimension",
         )
-        # `masked_fill` allows cpu scalar to be moved to cuda and xpu but not otherwise.
-        is_cpu_scalar = a.device.type in ["cuda", "xpu"] and value.device.type == "cpu"
-        torch._check(
-            is_cpu_scalar or value.device == a.device,
-            lambda: "Expected `value` to be on same device as `a`",
-        )
+        # CPU scalars are not supported for MPS in certain cases.
+        if a.device.type != "mps":
+            torch._check(
+                    (value.device.type == "cpu") or (value.device == a.device),
+                    lambda: "Expected `value` to be on same device as `a`",
+                )
```

Comment:
Comment and code do not agree? 
The code skips the validation for mps, but comment says mps should fail.

Reply:
The logic is the same as before for MPS, it was the exception it was skipping, just not explicitly. 

How about 
```
# Backends other than MPS allow CPU scalar to be moved to non-CPU device. Check scalar is on CPU, or on the same device as `a`.
```
Does that make more sense?